### ISSUE TITLE
Sometimes when I try to download an attachment it shows a 404 error

### ISSUE THREAD

**Issue by Robby2023** [Inevio/Gmail#60]
Sometimes when I click on download file it shows this:

![image](https://user-images.githubusercontent.com/1384919/34993402-8cee1394-fad0-11e7-8c85-b1f11c3bf646.png)

**Comment by burhan-project-manager** [comment]
@Robby2023 we have tested it with different attachments of existing emails but couldn't find error.
However when we send a fresh email with attachment, Gmail sometimes show 404 error but after some time if we hit that URL again, it downloads file successfully. May be it takes some time for Gmail to upload file?

**Comment by Robby2023** [comment]
Nope, I'm having the issue with an email I received yesterday and every time I try to download any of the attachments it shows me a 404: http://wz.io/f/Jg3pHkPT

**Comment by burhan-project-manager** [comment]
Can you please show hover URL of that attachment?

**Comment by Robby2023** [comment]
![hover url](https://user-images.githubusercontent.com/1384919/35281590-8ad74f98-0053-11e8-8c19-7029fb88c29e.png)

**Comment by burhan-project-manager** [comment]
@Robby2023 we are not able to find issue because we are not facing that issue any more.
Can you please forward that email so that we can debug?

**Comment by Robby2023** [comment]
Hello @burhan-project-manager, still having this issue, I'm forwarding you an email which attachment throws a 404

**Comment by burhan-project-manager** [comment]
I am able to download that file! 

![image](https://user-images.githubusercontent.com/28291997/35605214-132af064-0669-11e8-814f-d819c8092b3e.png)

May be it is true?

> @Robby2023 we have tested it with different attachments of existing emails but couldn't find error.
> However when we send a fresh email with attachment, Gmail sometimes show 404 error but after some time if we hit that URL again, it downloads file successfully. May be it takes some time for Gmail to upload file?

**Comment by jorge9200** [comment]
it's happening with all my mails with attachments when I press download to PC

**Comment by jorge9200** [comment]
[Feb 2 2018 10_21 PM.webm.zip](https://github.com/Inevio/Gmail/files/1690877/Feb.2.2018.10_21.PM.webm.zip)

**Comment by javiergarmon** [comment]
Please, use this endpoint instead the Gmail one:

```js
'https://download.horbito.com/gmail/' + accountId + '/' + encodeURIComponent( messageId ) + '/' + encodeURIComponent( attachmentId ) + '?name=' + encodeURIComponent( attachmentName )
```

**Comment by icastillejogomez** [comment]
@burhan-project-manager  let me know when it has been implemented please

**Comment by burhan-project-manager** [comment]
Implemented!

**Comment by burhan-project-manager** [comment]
It is downloading but immediately we get error. Don't know if it is network connectivity issue or something else?

![image](https://user-images.githubusercontent.com/28291997/36296723-8ce8330e-1310-11e8-99b2-8eb9288094c3.png)
